bats/tests/helpers/vm.bash

```diff
@@ -150,30 +150,65 @@ start_application() {
 
 container_engine_info() {
     run ctrctl info
-    assert_success
+    echo "$output"
+    assert_success || return
     assert_output --partial "Server Version:"
 }
 
 docker_context_exists() {
     run docker_exe context ls -q
-    assert_success
+    assert_success || return
     assert_line "$RD_DOCKER_CONTEXT"
 }
 
-buildkitd_is_running() {
-    run rdctl shell rc-service --nocolor buildkitd status
-    assert_success
-    assert_output --partial 'status: started'
+assert_service_status() {
+    local service_name=$1
+    local expect=$2
+
+    run rc_service "$service_name" status
+    # Some services (e.g. k3s) report non-zero status when not running
+    if [[ $expect == started ]]; then
```

Comment:
Hmmm, I'd almost like us to always quote constant strings, even if they match an identifier syntax, because the syntax for calling a function is the same as referencing a constant bareword.

Reply:
This is inside `[[ ... ]]`, so no function will be called unless you write `$(function)`. There is also no whitespace splitting or globing going on inside `[[ ... ]]`.

We run `shellcheck` (and `shfmt`) over all the BATS tests, so we'll detect if there are missing quotes. Run `make lint` inside the `bats/` directory before pushing your branch.

I plan on writing a "best practices and style guide" for our tests soon for recommendations beyond what is enforced by our code formatter/linters.

Reply:
Right, I wasn't aware that you can't call functions as barewords in `[[...]]` brackets

Reply:
> Right, I wasn't aware that you can't call functions as barewords in `[[...]]` brackets

It doesn't even work in `[ ... ]` brackets either. Because it is just calling `/bin/[` and everything after the bracket is an argument to the program. That's why you need to have a space after `[`; otherwise it would be part of the program name.

But inside `[ ... ]` you have to worry about white-space splitting and globing, which doesn't happen in `[[ ... ]]`.